### PR TITLE
Added Ecuador ID (IC) validator

### LINKED ISSUE #103: [v1.3.0] Implement Ecuador ID 🇪🇨  (CI) validator
- Implement the Ecuador ID validator
- Write tests

### PR DESCRIPTION
**Pull request by gheleri**
resolve #103 
It includes the validator and unit tests.

### CODE REVIEW

File: src/Core/SouthAmerica/Ecuador/EcuadorIdValidator.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * @package Reducktion\Socrates\Core\SouthAmerica\Ecuador
+ */
+class EcuadorIdValidator implements IdValidator
+{
+    public function validate(string $id): bool
+    {
+        $id = $this->sanitize($id);
+
+        $provinceCode = (int)substr($id, 0, 2);
+        $thirdDigit = (int)substr($id, 2, 1);
```
Comment (AlexOlival) [inline src/Core/SouthAmerica/Ecuador/EcuadorIdValidator.php:25]: If we're plucking a single digit, we can use PHP's array access (which I dearly miss in other languages :zany_face:)

File: src/Core/SouthAmerica/Ecuador/EcuadorIdValidator.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+            if ($posValue >= 10) {
+                $posValue = (int)array_sum(str_split($posValue));
+            }
+
+            $sum += $posValue;
+        }
+
+        $remaining = $sum % 10;
+        $result = $remaining !== 0 ? 10 - $remaining : 0;
```
Comment (AlexOlival) [inline src/Core/SouthAmerica/Ecuador/EcuadorIdValidator.php:76]: "remainder" is more appropriate here

**Review by AlexOlival — CHANGES_REQUESTED** [review]
Just two small tidbits. Great work again :hugs: 
Are there not many extractors in the Americas? That's a shame :stuck_out_tongue:

**Comment by gheleri** [comment]
> Are there not many extractors in the Americas? That's a shame

There're no rules in no man's land :P. Some countries are nearly impossible to get some info about IDs on internet

**Review by AlexOlival — APPROVED** [review]

**Review by JoaoFSCruz — APPROVED** [review]

**Pull request merged**